server-spi-private/src/main/java/org/keycloak/userprofile/DefaultAttributes.java

```diff
@@ -168,7 +168,9 @@ public AttributeMetadata getMetadata(String name) {
 
     @Override
     public Map<String, List<String>> getReadable() {
-        Map<String, List<String>> attributes = new HashMap<>(user.getAttributes());
```

Comment:
This method should not be called when the user is not available. Perhaps just return null at the beginning and avoid creating the hash map.